Extract reference for operator DIV to standalone header

ddavis-2015 opened the pull request:
Move the reference implementation to its own header so that micro
can use it without the unrelated depedencies of reference_ops.h.

This PR is part of the work to port operator DIV from lite to micro,
as tracked in Issue #45431.

ddavis-2015 committed:
Extract reference for operator DIV to standalone header

Move the reference implementation to its own header so that micro
can use it without the unrelated depedencies of reference_ops.h.

PR step 2 for issue micro: port op DIV from lite #45431

ddavis-2015 committed:
Merge branch 'master' into div-pr2

petewarden commented on tensorflow/lite/kernels/internal/reference/reference_ops.h:
```diff
@@ -25,6 +25,7 @@ limitations under the License.
 #include <limits>
 #include <memory>
 #include <type_traits>
+#include <vector>
```
Why is this new header needed? I don't see any usage of std::vector in the changed code.

ddavis-2015 replied:
As per discussion with @petewarden, this include was not needed as some cpplint include warnings can be ignored.  A new commit will be made to fix this.

ddavis-2015 committed:
Update to fix PR review problems

Remove <vector> include as per PR review and discussion with @petewarden

Remove profiler ScopeLabel as per discussion with @petewarden
